### DISCUSSION TITLE
How to show dap log under the dap bar window

### DISCUSSION THREAD

**Discussion by sufar** [mfussenegger/nvim-jdtls#430] [Q&A]
When debug a spring project started, but i can not see the dap logs:
![image](https://user-images.githubusercontent.com/82210712/221197145-a83b928e-7ef2-4149-9454-5945e2d3d946.png)

How to config the nvim-jdtls can like this dap logs look like: [nvim-metals](https://github.com/scalameta/nvim-metals)

![image](https://user-images.githubusercontent.com/82210712/221199430-5cec56aa-06b0-41f0-a117-ee38e94ea819.png)

[My nvim-jdtls java config file](https://github.com/sufar/nvim/blob/main/lua/lsp/java.lua)

[My nvim-metals config file](https://github.com/sufar/nvim/blob/main/lua/lsp/scala.lua)

**Comment by sufar** [comment]
It seems to caused by the [nvim-dap-ui](https://github.com/rcarriga/nvim-dap-ui)

[My nvim-dap-ui config file](https://github.com/sufar/nvim/blob/main/lua/core/dapui.lua)

I see a solution: [Add option to choose nvim-dap console type](https://github.com/mfussenegger/nvim-jdtls/issues/240)

I edit the local `~/.local/share/nvim/site/pack/packer/start/nvim-jdtls/lua/jdtls/dap.lua`
https://github.com/mfussenegger/nvim-jdtls/blob/e2a80d5f51c326d6623979d614dd7968ef752061/lua/jdtls/dap.lua#L485

change to:
```
console = 'internalConsole'; 
```

It works well:
![image](https://user-images.githubusercontent.com/82210712/222688004-d350dcd5-9ffd-412c-8b1c-6b572dfd4fe7.png)

But I don't known where and how to set this:
```
for _, config in pairs(dap.java.configurations) do
  config.console = 'internalConsole'
end
```

Can you help me?

**Reply by EpsilonKu** [reply]
I guess you are using `repl` not `console` in `dap-ui`. My spring project shows logs.
